examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/decoder.py

```diff
@@ -289,44 +291,60 @@ def run(self, specs, value, cluster_name: str, typename: str, array: bool):
                 field_name = field.name
                 field_type = field.data_type.name
                 field_array = field.is_list
-                # chip-tool returns the field code as an integer but the test suite expects
-                # a field name.
-                # To not confuse the test suite, the field code is replaced by its field name
-                # equivalent and then removed.
+
+                target_name = field_name
+
                 if str(field_code) in value:
-                    value[field_name] = self.run(
-                        specs,
-                        value[str(field_code)],
-                        cluster_name,
-                        field_type,
-                        field_array
+                    # chip-tool returns the field code as an integer but the test suite expects
+                    # a field name.
+                    # To not confuse the test suite, the field code is replaced by its field name
+                    # equivalent and then removed.
+                    target_name = str(field_code)
+                else:
+                    # darwin-framework-tool returns the field name but with a different casing than
+                    # what the test suite expects.
+                    # To not confuse the test suite, the field name is replaced by its field name
+                    # equivalent from the spec and then removed.
+
+                    if field_name == 'Description' and 'descriptionString' in value:
+                        # "Description" is returned as "descriptionString" since 'description' is a reserved keyword
+                        # and can not be exposed as if in an objc struct.
+                        target_name = 'descriptionString'
+
+                    # If field_name starts with a sequence of capital letters (which can happen for
+                    # output of asLowerCamelCase if the original string started that way) lowercase all but the last one.
```

Comment:
Is this the output of asLowerCamelCase, or the output of asUpperCamelCase?  Here `target_name` is the thing from the "spec" bits, right?  So I would not expect it to be the output of asLowerCamelCase.